blosc2/schunk.py

```diff
@@ -929,6 +1023,8 @@ def __dealloc__(self):
         super().__dealloc__()
 
 
+@_inherit_doc_parameter(SChunk.__init__, "mmap_mode:", {r"\* - 'w\+'[^*]+": ""})
+@_inherit_doc_parameter(SChunk.__init__, "initial_mapping_size:", {r"r\+ w\+, or c": "r+ or c"})
```

Comment:
I first thought about making a link from `open` to `SChunk.__init__` which would definitively be the easiest option^^

However, I am not much a fan of hyperlinks to other functions for reading the parameter documentation of the _current_ function. This always breaks the flow for the user since, usually, one wants to know what the current function is capable of without leaving the page. For some functions, there is even another level of indirection, e.g. if I want to know the possible arguments of `blosc2.asarray`, I first have to follow the documentation to `blosc2.empty` and then again to the documentation to `SChunk.__init__`.

This decorator is an intend to fix this problem without copying documentation manually. Additionally, I also alter the documentation here (the `w+` argument is removed because it cannot be used with `blosc2.open`).

I am not saying that the decorator is the best option btw. I think this is an inherent disadvantage of sphinx that it has no built-in mechanism for these use cases. A sphinx extensions might be better suited but I have no experience there and don't know whether this is even possible. On the other hand, with the decorator, `help(blosc2.open)` works properly.

What do you think?

Reply:
Yeah. Ok, let's do it.  I don't entirely like the fact that in most parts of docs we use an indirection, while in others, the docs will be just duplicated; but as this should go mostly unnoticed by users, that's fine, I suppose.